pandas/_libs/groupby.pyx

```diff
@@ -455,7 +455,7 @@ def _group_add(complexfloating_t[:, :] out,
     if len(values) != len(labels):
         raise ValueError("len(index) != len(labels)")
 
-    nobs = np.zeros((len(out), out.shape[1]), dtype=np.int64)
+    nobs = np.zeros((<object>out).shape, dtype=np.int64)
```

Comment:
What does the object cast buy us here?

Reply:
https://groups.google.com/forum/#!topic/cython-users/xheRGGOiuYw

For reasons unknown, `out.shape` is evaluated as an 8-tuple instead of 2-tuple, and this call raises at runtime without the `<object>` cast.  That's the reason why the status quo doesn't use `out.shape` directly on L458.

Reply:
Pretty interesting...and cast to object here doesn't trigger any underlying functions? Not sure if it just changes the type of pointer Cython uses or if it actually changes the memview